core/src/test/java/nl/nn/adapterframework/util/MiscTest.java

```diff
@@ -1,43 +1,610 @@
+                "    <p>This domain is for use in illustrative examples in documents. You may use this\n" +
+                "    domain in literature without prior coordination or asking for permission.</p>\n" +
+                "    <p><a href=\"https://www.iana.org/domains/example\">More information...</a></p>\n" +
+                "</div>\n" +
+                "</body>\n" +
+                "</html>\n");
+    }
+
+    /**
+     * Method: resourceToString(URL resource, String endOfLineString, boolean xmlEncode)
+     */
+    @Test
+    public void testResourceToStringForResourceEndOfLineStringXmlEncode() throws Exception {
+        URL resource = new URL("http://example.com/");
+        String s1 = Misc.resourceToString(resource, "end of the page", true);
+        assertEquals(s1, "&lt;!doctype html&gt;end of the page&lt;html&gt;end of the page&lt;head&gt;end of the page    &lt;title&gt;Example Domain&lt;/title&gt;end of the pageend of the page    &lt;meta charset=&quot;utf-8&quot; /&gt;end of the page    &lt;meta http-equiv=&quot;Content-type&quot; content=&quot;text/html; charset=utf-8&quot; /&gt;end of the page    &lt;meta name=&quot;viewport&quot; content=&quot;width=device-width, initial-scale=1&quot; /&gt;end of the page    &lt;style type=&quot;text/css&quot;&gt;end of the page    body {end of the page        background-color: #f0f0f2;end of the page        margin: 0;end of the page        padding: 0;end of the page        font-family: -apple-system, system-ui, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif;end of the page        end of the page    }end of the page    div {end of the page        width: 600px;end of the page        margin: 5em auto;end of the page        padding: 2em;end of the page        background-color: #fdfdff;end of the page        border-radius: 0.5em;end of the page        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);end of the page    }end of the page    a:link, a:visited {end of the page        color: #38488f;end of the page        text-decoration: none;end of the page    }end of the page    @media (max-width: 700px) {end of the page        div {end of the page            margin: 0 auto;end of the page            width: auto;end of the page        }end of the page    }end of the page    &lt;/style&gt;    end of the page&lt;/head&gt;end of the pageend of the page&lt;body&gt;end of the page&lt;div&gt;end of the page    &lt;h1&gt;Example Domain&lt;/h1&gt;end of the page    &lt;p&gt;This domain is for use in illustrative examples in documents. You may use thisend of the page    domain in literature without prior coordination or asking for permission.&lt;/p&gt;end of the page    &lt;p&gt;&lt;a href=&quot;https://www.iana.org/domains/example&quot;&gt;More information...&lt;/a&gt;&lt;/p&gt;end of the page&lt;/div&gt;end of the page&lt;/body&gt;end of the page&lt;/html&gt;end of the page");
+    }
```

Comment:
In stead of using a big concatinated string, it might be easier and more readable to use `TestFileUtils.getResource` and read the contents from a file instead

Reply:
done